src/librustc/ty/util.rs

```diff
@@ -525,6 +393,142 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
     }
 }
 
+struct TypeIdHasher<'a, 'gcx: 'a+'tcx, 'tcx: 'a> {
+    tcx: TyCtxt<'a, 'gcx, 'tcx>,
+    svh: &'a Svh,
```

Comment:
Probably we should just be hashing the crate-name/crate-discriminant pair, rather than employing the SVH.